internal/kconfig/kconfig.go

```diff
@@ -39,10 +39,10 @@ func Find() (*os.File, error) {
 }
 
 // Parse parses the kconfig file for which a reader is given.
-// All the CONFIG_* set will be put in the returned map as key with their
-// corresponding value as map value.
+// All the CONFIG_* which are in filteredConfigs and which are set set will be
+// put in the returned map as key with their corresponding value as map value.
 // If the kconfig file is not valid, error will be returned.
-func Parse(source io.ReaderAt) (map[string]string, error) {
+func Parse(source io.ReaderAt, filteredConfigs map[string]any) (map[string]string, error) {
```

Comment:
`map[string]any` is kind of convenient for where we are trying to call this function from, but it's not good API since it doesn't really document itself.

I'd prefer `map[string]struct{}` or `map[string]bool` even though it means another map allocated in the caller.